src/components/Products/Products.component.jsx

```diff
@@ -2,24 +2,35 @@ import { React, useState, useEffect } from "react";
 import Product from "../Product/Product.component";
 import ReactPaginate from "react-paginate";
 
-function Products({ header, data, categories, withDescription, itemsPerPage=20 }) {
+function Products({
```

Comment:
For this component, I think you are not using the API. It's necessary to start using it because it is going to change how your pagination works. As the API returns you the total number of pages, the page you are currently on and the endpoint for the next page. It does not return all elements, just the ones on the current page.

Reply:
Yeah, currently is only working with the fetched data received from its parent component, I will try to use a fetch on change of the selected categories and update this component accordingly.

Reply:
I refactored a bunch of stuff, including this component, now the pagination and the category filtering fetch new results from the API, it looks much better now!